docs/getting_started/setup_your_environment.md

```diff
@@ -144,6 +144,30 @@ project (`npm init -y` as necessary), then add the following block to your
 }
 ```
 
+#### LSP clients
+
+Deno has builtin support for the
+[Language server protocol](https://langserver.org).
+
+If your editor supports the LSP, you can use Deno as a language server for
+TypeScript and JavaScript.
+
+The editor can start the server with `deno lsp`.
+
+##### Example for Kakoune
+
+After installing the [`kak-lsp`](https://github.com/kak-lsp/kak-lsp) LSP client
+you can add the Deno language server by adding the following to your
+`kak-lsp.toml`
+
+```toml
+[language.deno]
+filetypes = ["typescript", "javascript"]
```

Comment:
```suggestion
filetypes = ["typescript", "typescriptreact", "javascript", "javascriptreact"]
```

Reply:
Kakoune uses `typescript` and `javascript` for `.jsx` and `.tsx` files.
https://github.com/mawww/kakoune/blob/ce96c3f26246ca1d1ae380e5a14e2880256ded25/rc/filetype/javascript.kak#L4-L10